cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -149,12 +152,70 @@ func generateInitramfsMounts() (err error) {
 // generateMountsMode* is called multiple times from initramfs until it
 // no longer generates more mount points and just returns an empty output.
 func generateMountsModeInstall(mst *initramfsMountsState) error {
+	// in install mode, we need to set the time to whatever timestamp we have
+	// in the initrd filesystem, using the /var/lib/systemd/timesync/clock file
+	// whose mtime is bumped every time the initrd is rebuilt
+	haveKernelInitrdClockTime := false
+	kernelInitrdClockTime := time.Time{}
+	currentSysTime := timeNow()
+	st, err := os.Stat(filepath.Join(dirs.GlobalRootDir, "/var/lib/systemd/timesync/clock"))
```

Comment:
Since this will be added in core-initrd, it will also add a service to push clock to the stampfile date. So I see this bit as redundant.